src/lpython/python_evaluator.cpp

```diff
@@ -119,10 +122,40 @@ Result<PythonCompiler::EvalResult> PythonCompiler::evaluate(
     }
 
     bool call_run_fn = false;
-    std::string return_type;
-    if (m->get_return_type(run_fn) != "none") {
-        call_run_fn = true;
-        return_type = m->get_return_type(run_fn);
+    std::string return_type = m->get_return_type(run_fn);
+    if (return_type != "none") {
+      call_run_fn = true;
+    }
+
+    llvm::Type *type = nullptr;
+    ASR::symbol_t *f = symbol_table->get_symbol("_" + run_fn);
+    if ((return_type == "struct") && (f)) {
```

Comment:
Slightly confused here. What are we processing inside this `if` and why do we need this processing?